src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSCloud.java

```diff
@@ -122,6 +119,12 @@ public ECSCloud(String name,
                 template.setOwer(this);
             }
         }
+
+        if (instanceCapStr.isEmpty()) {
+            this.instanceCap = Integer.MAX_VALUE;
+        } else {
+            this.instanceCap = Integer.parseInt(instanceCapStr);
```

Comment:
Should probably have a try/catch here to provide a default value if the string typed in isn't a valid integer.